sdk/core/azure-core/src/main/java/com/azure/core/util/Contexts.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.implementation.http.HttpPipelineCallContextHelper;
+
+import java.util.Objects;
+
+/**
+ * A utility type that can be used to add and retrieve instances commonly used in {@link Context}.
+ */
+public final class Contexts {
```

Comment:
I am considering if we need this additional type because it's not easily discoverable. `Context` is visible because it is present in all the `withResponse` methods of sync client. So, we can move the known key-value setters to `Context` itself.

Reply:
I generally don't worry too much about the discoverability of this type. It falls into the category of 'support type' that generally seems common in Java, e.g. Collections, StreamSupport, etc.

Reply:
I am fine with this new type. We should update documentation and probably replace `Context.NONE` in code snippets with `Contexts.empty()` to guide users toward this type.